src/Language/PureScript/Errors.hs

```diff
@@ -1306,6 +1316,25 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
         , line "Instance declarations of this type class are disallowed."
         ]
 
+    renderSimpleErrorMessage UnsupportedRoleDeclaration =
+      line $ "Role declarations are only supported for data types, not for type synonyms nor type classes."
+
+    renderSimpleErrorMessage (RoleDeclarationArityMismatch name expected actual) =
+      line $ T.intercalate " "
+        [ "The type"
+        , markCode (runProperName name)
+        , "expects"
+        , T.pack (show expected)
+        , if expected > 1 then "arguments" else "argument"
+        , "but its role declaration list"
+            <> if actual > expected then "" else " only"
```

Comment:
Adding "only" when there are not enough roles is a nice touch :+1:

Reply:
This is something I picked up from the `ClassInstanceArityMismatch` error ^^